src/main/extras/app/entrypoint.sh

```diff
@@ -108,35 +117,45 @@ FLAGS=(
     "-Dcom.sun.management.jmxremote.rmi.port=$CONTAINER_JFR_RMI_PORT"
     "-Djavax.net.ssl.trustStore=$SSL_TRUSTSTORE"
     "-Djavax.net.ssl.trustStorePassword=$SSL_TRUSTSTORE_PASS"
-    "-Djavax.net.ssl.keyStore=$SSL_KEYSTORE"
-    "-Djavax.net.ssl.keyStorePassword=$SSL_KEY_PASS"
 )
 
-if [ -z "$CONTAINER_JFR_RJMX_AUTH" ]; then
-    # default to true. This should never be disabled in production deployments
-    CONTAINER_JFR_RJMX_AUTH=true
-fi
-
 createSslStores
 importTrustStores
-generateSslCert
-if [ "$CONTAINER_JFR_RJMX_AUTH" = "true" ] || [ -n "$CONTAINER_JFR_RJMX_USER" ] ||
-    [ -n "$CONTAINER_JFR_RJMX_PASS" ]; then
-    createJmxCredentials
 
+if [ "$CONTAINER_JFR_DISABLE_JMX_AUTH" = "true" ]; then
+    banner "JMX Auth Disabled"
+    FLAGS+=("-Dcom.sun.management.jmxremote.authenticate=false")
+else
+    createJmxCredentials
     FLAGS+=("-Dcom.sun.management.jmxremote.authenticate=true")
     FLAGS+=("-Dcom.sun.management.jmxremote.password.file=$PWFILE")
     FLAGS+=("-Dcom.sun.management.jmxremote.access.file=$USRFILE")
-    FLAGS+=("-Dcom.sun.management.jmxremote.ssl=true")
-    FLAGS+=("-Dcom.sun.management.jmxremote.registry.ssl=true")
-    FLAGS+=("-Dcom.sun.management.jmxremote.ssl.need.client.auth=true")
-else
-    FLAGS+=("-Dcom.sun.management.jmxremote.authenticate=false")
+fi
+
+if [ "$CONTAINER_JFR_DISABLE_SSL" = "true" ]; then
+    banner "SSL Disabled"
     FLAGS+=("-Dcom.sun.management.jmxremote.ssl=false")
     FLAGS+=("-Dcom.sun.management.jmxremote.registry.ssl=false")
+else
+    FLAGS+=("-Dcom.sun.management.jmxremote.ssl.need.client.auth=true")
+
+    if [ -z "$KEYSTORE_PATH" ] || [ -z "$KEYSTORE_PASS" ]; then
+        generateSslCert
+        banner "Using self-signed SSL certificate"
+
+        KEYSTORE_PATH="$SSL_KEYSTORE"
+        KEYSTORE_PASS="$SSL_KEY_PASS"
+    fi
+
+    FLAGS+=("-Djavax.net.ssl.keyStore=$SSL_KEYSTORE")
+    FLAGS+=("-Djavax.net.ssl.keyStorePassword=$SSL_KEY_PASS")
```

Comment:
Should these be set to `KEYSTORE_PATH` and `KEYSTORE_PASS`?

Reply:
I think you're right. Fixed, thanks